pkg/container/reader.go

```diff
@@ -63,13 +64,34 @@ func (ctn Reader) GetAllDelegations() iter.Seq2[cid.Cid, *delegation.Token] {
 
 // GetInvocation returns the first found invocation.Token.
 // If none are found, ErrNotFound is returned.
+// If more than one invocation exist, ErrMultipleInvocations is returned.
 func (ctn Reader) GetInvocation() (*invocation.Token, error) {
+	var res *invocation.Token
 	for _, t := range ctn {
 		if inv, ok := t.(*invocation.Token); ok {
-			return inv, nil
+			if res != nil {
+				return nil, ErrMultipleInvocations
+			}
+			res = inv
+		}
+	}
+	if res == nil {
+		return nil, ErrNotFound
+	}
+	return res, nil
+}
+
+// GetAllInvocations returns all the invocation.Token in the container.
+func (ctn Reader) GetAllInvocations() iter.Seq2[cid.Cid, *invocation.Token] {
```

Comment:
I love the new iterators - so functional and satisfyingly lazy like Java Stream.